File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.*;

class Solution {
    static ArrayList<String> list;
    static String[] words = {"A","E","I","O","U"};
    
    public int solution(String word) {
        int answer = 0;
        list = new ArrayList<>();
        dfs("",0);
        
        for(int i=0; i<list.size(); i++){
            if(word.equals(list.get(i))){
                answer = i;
                break;
            }
        }
        
        return answer;
    }
    
    private static void dfs(String str, int len){
        list.add(str);
        if(len == 5) return;
        
        for(int i=0; i<5; i++){
            dfs(str+words[i],len+1);
        }
    }
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
from itertools import product

def solution(word):
    answer = 0
    vowels = ['A', 'E', 'I', 'O', 'U']
    words = []
    
    for length in range(1,6):
        for p in product(vowels, repeat=length):
            words.append(''.join(p))
    
    words.sort()
    
    return words.index(word) + 1